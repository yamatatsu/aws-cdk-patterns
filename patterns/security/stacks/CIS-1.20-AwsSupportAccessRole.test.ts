import { SynthUtils } from "@aws-cdk/assert"
import { App } from "aws-cdk-lib"
import { Cis_1_20_AwsSupportAccessRole } from "./CIS-1.20-AwsSupportAccessRole"

test("snapshot test", () => {
  const app = new App()

  const target = new Cis_1_20_AwsSupportAccessRole(app, "Target")

  expect(SynthUtils.toCloudFormation(target)).toMatchSnapshot()
})
