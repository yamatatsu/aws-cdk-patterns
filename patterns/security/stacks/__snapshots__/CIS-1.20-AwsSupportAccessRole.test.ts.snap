// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
Object {
  "Resources": Object {
    "Role1ABCC5F0": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "AWS": Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":iam::",
                      Object {
                        "Ref": "AWS::AccountId",
                      },
                      ":root",
                    ],
                  ],
                },
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "Description": "https://docs.aws.amazon.com/securityhub/latest/userguide/securityhub-cis-controls.html#cis-1.20-remediation",
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/AWSSupportAccess",
              ],
            ],
          },
        ],
        "RoleName": "AWSSupportAccessRole",
      },
      "Type": "AWS::IAM::Role",
    },
  },
}
`;
